docs/developer-docs/setup/best-practices/troubleshooting.mdx

```diff
@@ -0,0 +1,28 @@
+# Troubleshooting latency 
+
+## Overview
+
+When a dapp or canister is running with a high amount of latency, there are some steps that developers can take to troubleshoot or optimize the performance of their canisters. 
+
+This document will detail some troubleshooting methods and performance optimizations that developers can use when troubleshooting latency. 
+
+## Troubleshooting best practices 
+
+### Recommendation: Use query calls instead of update calls when possible.
+
+[Query calls](/docs/current/developer-docs/backend/design-dapps#using-query-calls) are executed on a single node within a subnet and do not go through consensus, therefore they can be returned much faster than an update call. For some applications, query calls can be used in place of update calls to reduce latency. 
```

Comment:
We can consider adding this note from Andriy's blog post: https://internetcomputer.org/blog/features/improving-query-latencies#takeaways-how-to-improve-query-latencies

Reply:
Added a section on efficient query calls